community/kernel/src/main/java/org/neo4j/kernel/impl/store/format/RecordFormats.java

```diff
@@ -44,9 +48,15 @@ public Factory( String key, String... altKeys )
 
     String storeVersion();
 
+    /**
+     * @return format generation, with the intent of usage being that a store can migrate to a newer or
+     * same generation, but not to an older generation.
+     */
+    int generation();
```

Comment:
since our formats has version as part of class name we can have same version inside of them, with exactly same value stored in a version field and do comparison based on those instead